src/query/plan/rewrite/join.hpp

```diff
@@ -59,6 +59,12 @@ class JoinRewriter final : public HierarchicalLogicalOperatorVisitor {
 
     ExpressionRemovalResult removal = RemoveExpressions(op.expression_, filter_exprs_for_removal_);
     op.expression_ = removal.trimmed_expression;
+    if (op.expression_) {
+      Filters leftover_filters;
+      leftover_filters.CollectFilterExpression(op.expression_, *symbol_table_);
+      op.all_filters_ = leftover_filters;
```

Comment:
```suggestion
      op.all_filters_.CollectFilterExpression(op.expression_, *symbol_table_);
```

Reply:
I think all_filters_ variable should now be renamed since it doesn't store all filters. Also, I think `CollectFilterExpression` will not necessarily always collect leftover filters?

Reply:
Do you even need `all_filters`?

Reply:
that's a good observation, I may not need the filters, I'll see

Reply:
I need the filters since we implemented that in a way to parse easily the information for the operator. At this point I'm just fixing the bug of wrong printing but I think it makes sense to remove all filters totally and parse the expression in a visitor rather than having something which needs invalidation if changed.

Reply:
Also a comment for @as51340 : I need to reassign the operator's all_filters_ since when we apply the indexing we don't need to just add new filters but to reassign that variable which has changed. That's why I go into the op.expression again and reassign it. I think what `CollectFilterExpression` does is append to the list of already present filters

Reply:
Okay. But can we change the code as in the suggestion or not?